src/pset/map/global.rs

```diff
@@ -426,14 +473,21 @@ impl Decodable for Global {
             }
         }
 
-        if let Some(tx) = tx {
-            let mut rv: Global = Global::from_unsigned_tx(tx)?;
-            rv.version = version.unwrap_or(0);
-            rv.xpub = xpub_map;
-            rv.unknown = unknowns;
-            Ok(rv)
-        } else {
-            Err(Error::MustHaveUnsignedTx.into())
-        }
+        // Mandatory fields
+        let version = version.ok_or(Error::MissingPsetVersion)?;
```

Comment:
We should require the version to be 2, not just that it be present.

Reply:
Sure, will fix this. I was thinking about future versions.